### PR TITLE
As per email, plus a couple questions the snuck in

### PR DESCRIPTION
**Pull request by mslinn**

### CODE REVIEW

File: modules/auth/auth-core/src/main/java/auth/core/module/SilhouetteContext.java [hunk 1]
```diff
@@ -10,6 +10,7 @@
 import static java.lang.annotation.ElementType.PARAMETER;
 import static java.lang.annotation.RetentionPolicy.RUNTIME;
 
+// TODO what is this for?
 @BindingAnnotation
```
Comment (KadekM) [inline modules/auth/auth-core/src/main/java/auth/core/module/SilhouetteContext.java:14]: So I can tell Guice that my class depends on instance (execution context) marked by this attribute

File: modules/auth/auth-core/src/main/scala/auth/core/module/SilhouetteModule.scala [hunk 2]
```diff
@@ -25,7 +25,7 @@ sealed class SilhouetteModule extends AbstractModule with ScalaModule {
     bind[IDGenerator].toProvider[IDGeneratorProvider]
 
     bind[ExecutionContext]
-      .annotatedWith[SilhouetteContext]
+      .annotatedWith[SilhouetteContext] // TODO what does this do?
```
Comment (KadekM) [inline modules/auth/auth-core/src/main/scala/auth/core/module/SilhouetteModule.scala]: This is how I tell it - any EC that is annotated by SilhouetteContext should be provided by  that belowmentioned provider.

I.e.
https://github.com/KadekM/play-slick-silhouette-auth-api/blob/master/modules/auth/auth-core/src/main/scala/auth/core/module/SilhouetteModule.scala#L39

Lots of silhouette classes require execution context, and instead of running everything on  play's EC, I have decided to provide separate configurable thread context, where hopefully all sillhouette-related stuff happens.

So, in other words, if some client does heavy stuff on global ec, or play's one, silhouette-related stuff should not be bound by that.

**Comment by KadekM** [comment]
I'll merge it, but can you please first remove those `what is this for`s?

**Comment by mslinn** [comment]
Replaced questions with what I hope are informative comments

**Comment by KadekM** [comment]
Cool thanks!

**Comment by mslinn** [comment]
I notice the Guice bindings are made at runtime. Is there a reason why compile-time binding is not used?

**Comment by KadekM** [comment]
I find it most straightforward, also it's what silhouette does internally so I was being consistent.

**Comment by mslinn** [comment]
I don't know Guice well enough to know if there is a technical reason why Silhouette uses runtime bindings. Clearly, compile-time bindings are preferred if possible. I don't think consistency is as important as robustness. Why allow for a whole class of runtime failures if an alternative exists?

**Comment by mslinn** [comment]
http://mariussoutier.com/blog/2015/12/06/playframework-2-4-dependency-injection-di/#compile-time-di

**Comment by mslinn** [comment]
I did not find any instances of "Retention(RUNTIME)" in Silhouette, just [bind statements](https://github.com/mohiva/play-silhouette/blob/030608725c3f5f4abdf5dc171020a93c37a2a89b/silhouette/app/com/mohiva/play/silhouette/api/actions/UserAwareAction.scala#L182-L183). Looks like Silhouette only uses compile-time DI>
